server/medical-contact/collection.ts

```diff
@@ -0,0 +1,124 @@
+   *
+   * @param {string} medicalContactId - The id of the medical contact to find
+   * @return {Promise<HydratedDocument<Account>> | Promise<null>} - The medical contact with the given id, if any
+   */
+   static async findOne(medicalContactId: Types.ObjectId | string): Promise<HydratedDocument<MedicalContact>> {
+    return MedicalContactModel.findOne({_id: medicalContactId}).populate('ownerId');
+  }
+
+  /**
+   * Get all the medical contacts in by given owner
+   *
+   * @param {string} ownerId - The id of the owner of the medical contacts
+   * @return {Promise<HydratedDocument<MedicalContact>[]>} - An array of all of the medical contacts sorted in alphabetical order by last name
+   */
+   static async findAllByOwnerId(ownerId: string): Promise<Array<HydratedDocument<MedicalContact>>> {
+    const owner = await AccountCollection.findOneByAccountId(ownerId);
+    return MedicalContactModel.find({ownerId: owner._id}).sort({last_name: 1}).populate('ownerId');
+  }
+
+  /**
+   * Update medical contact's information
+   *
+   * @param {string} medicalContactId - The medicalContactId of the medical contact to update
+   * @param {Object} contactDetails - An object with the medical contact's updated details
+   * @return {Promise<HydratedDocument<MedicalContact>>} - The updated medical contact
+   */
+  static async updateOne(medicalContactId: Types.ObjectId | string, contactDetails: {title?: string; first_name?: string, last_name?: string, hospital?: string, specialty?: string, phone_number?: string, notes?: string}): Promise<HydratedDocument<MedicalContact>> {
+    
+    const medicalContact = await MedicalContactModel.findOne({_id: medicalContactId});
+    
+    if (contactDetails.title) {
+        medicalContact.title = contactDetails.title;
+    }
+
+    if (contactDetails.first_name) {
+        medicalContact.first_name = contactDetails.first_name;
+    }
+
+    if (contactDetails.last_name) {
+        medicalContact.last_name = contactDetails.last_name;
+    }
+
+    if (contactDetails.hospital) {
+        medicalContact.hospital = contactDetails.hospital;
+    }
+
+    if (contactDetails.specialty) {
+        medicalContact.specialty = contactDetails.specialty;
+    }
+
+    if (contactDetails.notes) {
+        medicalContact.notes = contactDetails.notes;
+    }
+
+    await medicalContact.save();
+    return medicalContact.populate('ownerId');
+  }
+
+  /**
+   * Delete a medical contact from the collection.
+   *
+   * @param {string} medicalContactId - The medicalContactId of medical contact to delete
+   * @return {Promise<Boolean>} - true if the user has been deleted, false otherwise
+   */
+  static async deleteOne(medicalContactId: Types.ObjectId | string): Promise<boolean> {
+
+    const medicalContact = await MedicalContactModel.findOne({_id: medicalContactId});
+    medicalContact.active = false; // Since deleted, they are no longer active
+    await medicalContact.save();
+
+    return medicalContact.active === false;
```

Comment:
you don't use the return value in your route, so I would just return void on this one.

Reply:
Good catch, thanks!